include/drivers/dai.h

```diff
@@ -0,0 +1,462 @@
+ * @retval -EINVAL Invalid argument.
+ * @retval -ENOSYS DAI_DIR_BOTH value is not supported.
+ */
+static inline int dai_config_set(const struct device *dev,
+				 const struct dai_config *cfg,
+				 const void *bespoke_cfg)
+{
+	const struct dai_driver_api *api = (const struct dai_driver_api *)dev->api;
+
+	return api->config_set(dev, cfg, bespoke_cfg);
+}
+
+/**
+ * @brief Fetch configuration information of a DAI driver
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param dir Stream direction: RX or TX as defined by DAI_DIR_*
+ * @retval Pointer to the structure containing configuration parameters,
+ *         or NULL if un-configured
+ */
+static inline const struct dai_config *dai_config_get(const struct device *dev,
+						      enum dai_dir dir)
+{
+	const struct dai_driver_api *api = (const struct dai_driver_api *)dev->api;
+
+	return api->config_get(dev, dir);
+}
+
+/**
+ * @brief Fetch properties of a DAI driver
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param dir Stream direction: RX or TX as defined by DAI_DIR_*
+ * @param stream_id Stream id: some drivers may have stream specific
+ *        properties, this id specifies the stream.
+ * @retval Pointer to the structure containing properties,
+ *         or NULL if error or no properties
+ */
+static inline const struct dai_properties *dai_get_properties(const struct device *dev,
+							      enum dai_dir dir,
+							      int stream_id)
+{
+	const struct dai_driver_api *api = (const struct dai_driver_api *)dev->api;
+
+	return api->get_properties(dev, dir, stream_id);
+}
+
+/**
+ * @brief Send a trigger command.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param dir Stream direction: RX, TX, or both, as defined by DAI_DIR_*.
+ *            The DAI_DIR_BOTH value may not be supported by some drivers.
+ *            For those, triggering need to be done separately for the RX
+ *            and TX streams.
+ * @param cmd Trigger command.
+ *
+ * @retval 0 If successful.
+ * @retval -EINVAL Invalid argument.
+ * @retval -EIO The trigger cannot be executed in the current state or a DMA
+ *         channel cannot be allocated.
+ * @retval -ENOMEM RX/TX memory block not available.
+ * @retval -ENOSYS DAI_DIR_BOTH value is not supported.
+ */
+static inline int dai_trigger(const struct device *dev,
+			      enum dai_dir dir,
+			      enum dai_trigger_cmd cmd)
+{
+	const struct dai_driver_api *api = (const struct dai_driver_api *)dev->api;
+
+	return api->trigger(dev, dir, cmd);
+}
+
+/**
+ * Configures timestamping in attached DAI.
+ * @param dev Component device.
+ * @param cfg Timestamp config.
+ *
+ * Optional method.
+ *
+ * @retval 0 If successful.
+ */
+static inline int dai_ts_config(const struct device *dev, struct dai_ts_cfg *cfg)
+{
+	const struct dai_driver_api *api = (const struct dai_driver_api *)dev->api;
+
+	if (!api->ts_config)
+		return -EINVAL;
```

Comment:
When a function is not implemented, `-ENOSYS` should be returned. See [API Design Guidelines](https://docs.zephyrproject.org/latest/reference/api/design_guidelines.html#return-codes).
Braces should also be added (see [Coding Style](https://docs.zephyrproject.org/latest/contribute/index.html#coding-style)).